src/profile/components/modals/deleteProfileError/DeleteProfileError.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { Button, Dialog, IconAlertCircle } from 'hds-react';
+import { useTranslation } from 'react-i18next';
+import { ApolloError } from '@apollo/client';
+
+import { getModalProps } from '../getModalProps';
+import profileConstants from '../../../constants/profileConstants';
+
+export type Props = {
+  error?: ApolloError | Error;
+  onClose: () => void;
+};
+
+function DeleteProfileError({
+  error,
+  onClose,
+}: Props): React.ReactElement | null {
+  const { t } = useTranslation();
+
+  if (!error) {
+    return null;
+  }
+  const getErrorTranslationKey = (resultError: Props['error']): string => {
+    if (!resultError) {
+      return 'deleteProfileErrorModal.genericError';
+    }
+    const graphQlErrors = (resultError as ApolloError).graphQLErrors;
+    if (graphQlErrors) {
+      const filteredErrors: string[] = graphQlErrors
+        .map(graphQlError => graphQlError?.extensions?.code)
+        .filter(graphQlError => graphQlError);
```

Comment:
Naming thoughts:

- `filteredErrors` could be named `errorCodes`, because that's what it seems to contain.
- `map` "extracts" the `code` from the errors, so the filter line could be: `.filter(code => code)`

Reply:
Fixed. I removed the array.filter as it is not necessary. array.includes works without it.